File: src/main/java/upbrella/be/store/dto/response/SingleImageUrlResponse.java
package upbrella.be.store.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import upbrella.be.store.entity.StoreImage;

@Getter
@Builder
@AllArgsConstructor
public class SingleImageUrlResponse {

    private long id;
    private String imageUrl;

    public static SingleImageUrlResponse createImageUrlResponse(StoreImage imageUrl) {

        return SingleImageUrlResponse.builder()
                .id(imageUrl.getId())
                .imageUrl(imageUrl.getImageUrl())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleImageUrlResponse.kt
package upbrella.be.store.dto.response

import upbrella.be.store.entity.StoreImage

data class SingleImageUrlResponse(
    val id: Long?,
    val imageUrl: String?, // 기존 호환성 유지
    val imageUrls: ImageUrlsResponse
) {
    companion object {
        fun createImageUrlResponse(storeImage: StoreImage): SingleImageUrlResponse {
            val imageUrlsResponse = storeImage.getImageUrlsResponse()
            return SingleImageUrlResponse(
                id = storeImage.id,
                imageUrl = storeImage.imageUrl, // 기존 호환성
                imageUrls = imageUrlsResponse
            )
        }
    }
}
